feat: adds account generation and import API functions (#1887)
ApeWorX/ape: docs/userguides/accounts.md

If you do not use the `--word-count` option, Ape will use the default word count of 12.
You can use all of these together or separately to control the way Ape creates and displays your account information.

This same functionality is also scriptable with the same inputs as the `generate` command:

```python

> from ape_acounts import generate_account

Reviewer: ```suggestion
from ape_accounts import generate_account
```

Author: Good catch

alias = "my-account"
passphrase = "mySecurePassphrase"
hd_path = "m/44'/60'/0'/0/0"

> word_count = 12

Reviewer: The first example I would just skip the kwargs, to demonstrate they are not necessary. Or just comment that you only need the first two args

Author: Yeah, I was torn on this.  Fewer args is more attractive but wanted to demonstrate the available kwargs.  Will this stuff get autodocced?  Can I link to the actual API?

Reviewer 2: You can link to the method from the autodoc, the user guide can just have a simple example and then be like "for more info on using this method, see the method's documentation" with that part as a link..

search around for markdown links that start with `../methoddocs/` to see how it is done

Author: ah, I see how the API is documented now. Though none of the core plugins are documented by autodoc currently.  Should I add one for ape_accounts?

Author: I added an autodoc page for ape_accounts.

> account, mnemonic = generate_account(alias, passphrase, hd_path, word_count)

Reviewer: Now that we are linking to the methoddocs, I think this is appropiate to drop the hd path/word count
```suggestion
account, mnemonic = generate_account(alias, passphrase)
```

If you use the `--hd-path` option, you will need to pass the [HDPath](https://help.myetherwallet.com/en/articles/5867305-hd-wallets-and-derivation-paths) you'd like to use as an argument in the command.
If you do not use the `--hd-path` option, Ape will use the default HDPath of (Ethereum network, first account).

> You can script account imports using a seed phrase:

Reviewer: ```suggestion
You can [import](add link here...) an account programmatically using a seed phrase:
```

Author: What would the link be to?

Author: Assuming it's the new autodoc page, I have one in there now.  Updated using your phrasing.

```python

> from ape_acounts import generate_account

Reviewer: ```suggestion
from ape_acounts import import_account_from_mnemonic
```

alias = "my-account"
passphrase = "mySecurePassphrase"
mnemonic = "test test test test test test test test test test test junk"
hd_path = "m/44'/60'/0'/0/0"

> account = import_account_from_mnemonic(alias, passphrase, mnemonic, hd_path)

Reviewer: Same here
```suggestion
account = import_account_from_mnemonic(alias, passphrase, mnemonic)
```

print(f'Your imported account address is: {account.address}')
```

> Or a raw private key:

Reviewer: ```suggestion
Or using a raw private key:
```
